Server/ReasnAPI/ReasnAPI/Controllers/EventsController.cs

```diff
@@ -1,171 +1,307 @@
+        if (existingEvent.OrganizerId != user.Id && user.Role != UserRole.Admin)
+        {
+            return Forbid();
+        }
+
+        if (existingEvent.Status != eventUpdateRequest.Status && user.Role != UserRole.Admin && eventUpdateRequest.Status != EventStatus.Cancelled)
+        {
+            return Forbid();
+        }
+        eventService.UpdateAddressForEvent(eventUpdateRequest.AddressDto, existingEvent.AddressId, slug);
+        eventService.UpdateEvent(existingEvent.Id, eventDto);
+
+        return Ok();
     }
 
     [HttpGet]
     [Authorize(Roles = "Admin")]
     [Route("requests")]
-    public IActionResult GetEventsRequests(string slug)
+    [ProducesResponseType<List<EventResponse>>(StatusCodes.Status200OK)]
+    public IActionResult GetEventsRequests()
     {
-        throw new NotImplementedException();
+        var events = eventService.GetEventsByFilter(e => e.Status == EventStatus.PendingApproval);
+       
+        return Ok(events); 
     }
 
     [HttpPost]
     [Authorize(Roles = "Admin")]
     [Route("{slug}/approve")]
-    public IActionResult ApproveEventRequest(string slug)
+    [ProducesResponseType<EventDto>(StatusCodes.Status200OK)]
+    public IActionResult ApproveEventRequest([FromRoute] string slug)
     {
-        throw new NotImplementedException();
+        var eventToApprove = eventService.GetEventBySlug(slug);
+
+        eventToApprove.Status = EventStatus.Approved;
+        eventService.UpdateEvent(eventToApprove.Id, eventToApprove.ToDto());
+        return Ok();
+    }
+
+    [HttpPost]
+    [Authorize(Roles = "Admin, Organizer")]
+    [Route("{slug}/images")]
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    public async Task<IActionResult> AddEventImage([FromRoute] string slug, [FromForm] List<IFormFile> images)
+    {
+        var @event = eventService.GetEventBySlug(slug);
+        var user = userService.GetCurrentUser();
+
+        if (@event.OrganizerId != user.Id && user.Role != UserRole.Admin)
+        {
+            return Forbid();
+        }
+
+        var imageDtos = new List<ImageDto>();
+
+        foreach (var formFile in images.Where(formFile => formFile.Length > 0))
+        {
+            using var ms = new MemoryStream();
+            await formFile.CopyToAsync(ms);
+            var fileBytes = ms.ToArray();
+
+            imageDtos.Add(new ImageDto
+            {
+                ObjectId = @event.Id,
+                ObjectType = ObjectType.Event,
+                ImageData = fileBytes
+            });
+        }
+
+        imageService.CreateImages(imageDtos);
+        return Ok();
+    }
+
+    [HttpPut]
+    [Authorize(Roles = "Admin, Organizer")]
+    [Route("{slug}/images")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<IActionResult> UpdateEventImage([FromRoute] string slug, [FromForm] List<IFormFile> images)
+    {
+        var user = userService.GetCurrentUser();
+        var @event = eventService.GetEventBySlug(slug);

```

Comment:
Can we use different name to not use reserved keywords?

Reply:
changed to relatedEvent for all ones